samples/snippets/fsharp/VS_Snippets_CLR/AnimalAttributes/FS/customattribute.fs

```diff
@@ -0,0 +1,41 @@
+﻿//<Snippet1>
+open System
+
+// An enumeration of animals. Start at 1 (0 = uninitialized).
+type Animal =
+    | Dog = 1
+    | Cat = 2
+    | Bird = 3
+
+// A custom attribute to allow a target to have a pet.
+type AnimalTypeAttribute(pet) =
+    inherit Attribute()
+    member val Pet = pet with get, set
+
+// A test class where each method has its own pet.
+type AnimalTypeTestClass() =
+    [<AnimalType(Animal.Dog)>]
+    member _.DogMethod() = ()
+
+    [<AnimalType(Animal.Cat)>]
+    member _.CatMethod() = ()
+
+    [<AnimalType(Animal.Bird)>]
+    member _.BirdMethod() = ()
+
+let testClass = AnimalTypeTestClass()
+let type' = testClass.GetType()
```

Comment:
Please don't use names ending with `'` - change to `theClassType` or `theType` or `typ`